clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -521,6 +521,28 @@ LogicalResult VecCreateOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// VecTernaryOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult VecTernaryOp::verify() {
+  // Verify that the condition operand is a vector of integral type.
+  if (!getCond().getType().getEltType().isa<mlir::cir::IntType>()) {
```

Comment:
You can create a "VectorOfInts" predicate in tablegen just like `ExceptionInfoPtr` in `CIRTypes.td`. And then use it to constraint `cond` in `VecTernaryOp`'s definition.

Reply:
Okay.  My next PR will be some cleanup and refactoring, including the `cir.vec.splat` operation that you previously suggested.  Can I do the `VectorOfInts` predicate in that PR?

Reply:
Sounds good